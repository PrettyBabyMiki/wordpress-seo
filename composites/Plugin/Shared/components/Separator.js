// External dependencies.
import styled from "styled-components";

// Internal dependencies.
import colors from "../../../../style-guide/colors";

/**
 * Styled component with the purpose to visually separate in the DOM
 *
 * @param {Object} props             The props.
 * @param {string} props.margin      The optional CSS margin.
 * @param {string} props.borderColor The optional CSS border color.
 *
 * @returns {ReactElement} The Separator styled component.
 */
export default styled.div`
	margin: ${ props => props.margin ? props.margin : "16px 0" };
	border: 0;
	border-bottom: 1px solid ${ props => props.borderColor ? props.borderColor : colors.$color_grey_light };
`;
