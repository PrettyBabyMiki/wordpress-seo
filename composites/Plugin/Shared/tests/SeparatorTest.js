import React from "react";
import renderer from "react-test-renderer";

import Separator from "../components/Separator";

describe( Separator, () => {
	it( "matches the snapshot", () => {
		const component = renderer.create(
			<Separator />
		);

		let tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	it( "matches the snapshot when passing a border color", () => {
		const component = renderer.create(
			<Separator borderColor={ "#ddd" } />
		);

		let tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	it( "matches the snapshot when passing a margin", () => {
		const component = renderer.create(
			<Separator margin={ "32px 6px" } />
		);

		let tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
} );
