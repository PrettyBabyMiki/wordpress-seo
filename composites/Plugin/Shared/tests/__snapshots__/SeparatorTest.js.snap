// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledComponent matches the snapshot 1`] = `
.c0 {
  margin: 16px 0;
  border: 0;
  border-bottom: 1px solid #f1f1f1;
}

<div
  className="c0"
/>
`;

exports[`StyledComponent matches the snapshot when passing a border color 1`] = `
.c0 {
  margin: 16px 0;
  border: 0;
  border-bottom: 1px solid #ddd;
}

<div
  className="c0"
/>
`;

exports[`StyledComponent matches the snapshot when passing a margin 1`] = `
.c0 {
  margin: 32px 6px;
  border: 0;
  border-bottom: 1px solid #f1f1f1;
}

<div
  className="c0"
/>
`;
